lib/nodeo/Antenna.js

```diff
@@ -0,0 +1,153 @@
+function array_factorcir(x1,phi,phi_desired,distance,dim){
+//As the function name signifies here the array factor is calculated 
+ 
+var pi=3.141592654;
+var  y=0;
+var  y1=0;
+//   for i1=1:dim/2
+//       delphi=2*pi*(i1-1)/dim;
+//       shi=cos(phi-delphi)-cos(phi_desired*(pi/180)-delphi);
+//       shi=shi*dim*distance;
+//       y=y+x1(i1)*cos(shi+x1(dim/2+i1)*(pi/180));
+//   end;
+//   y=y*2;
+//   y=abs(y);
+ 
+
+  //for i1=1:dim/2
+  for(var i1 = 0; i<dim/2;i1++){
+      var delphi=2*pi*(i1-1)/dim;
+      var shi=Math.cos(phi-delphi)-Math.cos(phi_desired*(pi/180)-delphi);
+      shi=shi*dim*distance;
+      y=y+x1[i1]*Math.cos(shi+x1[dim/2+i1]*(pi/180));
+  }
+   //for i1=dim/2+1:dim
+   for(var i1 = dim/2; i1<dim;i1++){
+      var delphi=2*pi*(i1-1)/dim;
+      var shi=cos(phi-delphi)-Math.cos(phi_desired*(pi/180)-delphi);
+      shi=shi*dim*distance;
+      y=y+x1[i1-dim/2]*Math.cos(shi-x1[i1]*(pi/180));
+  }
+  //for i1=1:dim/2
+  for(var i1 = 0; i1<dim/2;i1++){
+      var delphi=2*pi*(i1-1)/dim;
+      var shi=Math.cos(phi-delphi)-Math.cos(phi_desired*(pi/180)-delphi);
+      shi=shi*dim*distance;
+      y1=y1+x1[i1]*Math.sin(shi+x1[dim/2+i1]*(pi/180));
+  }
+  //for i1=dim/2+1:dim
+  for(var i1 = dim/2;i1<dim;i1++){
+      var delphi=2*pi*(i1-1)/dim;
+      var shi=Math.cos(phi-delphi)-Math.cos(phi_desired*(pi/180)-delphi);
+      shi=shi*dim*distance;
+      y1=y1+x1(i1-dim/2)*Math.sin(shi-x1(i1)*(pi/180));
+  }
+  y=y*y+y1*y1;
+  y=Math.sqrt(y);
+  
+return y;
+}
+
+
+
+
+function[y sllreturn bwfn]=antennafunccircular(x1,nullMatrix,phi_desired,distance)

```

Comment:
Line is too long.<br>Missing name in function declaration.<br>Expected '(' and instead saw '['.<br>Expected ')' to match '[' from line 54 and instead saw 'sllreturn'.<br>'function closure expressions' is only available in Mozilla JavaScript extensions (use moz option).<br>Expected an identifier and instead saw ']'.<br>Expected an operator and instead saw '='.<br>Expected an assignment or function call and instead saw an expression.<br>Missing semicolon.<br>Identifier 'phi_desired' is not in camel case.